app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -0,0 +1,361 @@
+    if (lastRunExecutorMemoryOverhead == 0) {
+      memoryOverhead = Math.max(suggestedExecutorMemory / 10, 384);
+    }
+
+    long roundedContainerSize = getRoundedContainerSize(suggestedExecutorMemory + memoryOverhead + FileUtils.ONE_MB);
+    long executorMemory =
+        (long) (suggestedExecutorMemory + (roundedContainerSize - suggestedExecutorMemory - memoryOverhead - FileUtils.ONE_MB) * 0.9);
+
+    return executorMemory;
+  }
+
+  /**
+   * This method is to get the rounded executor memory. We don't want to round executor memory to
+   * multiple of 1GB as container size is decided by executor memory + memoryOverhead. So round the
+   * executor memory + memoryOverhead and then distribute extra memory to executor memory
+   *
+   * @return rounded driver memory
+   */
+  private long getRoundedDriverMemory() {
+    long memoryOverhead = lastRunDriverMemoryOverhead;
+    if (lastRunDriverMemoryOverhead == 0) {
+      memoryOverhead = Math.max(suggestedDriverMemory / 10, 384);
+    }
+
+    long roundedContainerSize = getRoundedContainerSize(suggestedDriverMemory + memoryOverhead + FileUtils.ONE_MB);
+    long driverMemory =
+        (long) (suggestedDriverMemory + (roundedContainerSize - suggestedDriverMemory - memoryOverhead - FileUtils.ONE_MB) * 0.9);
+
+    return driverMemory;
+  }
+
+  /**
+   * Suggest executor memory based on peak JVM used memory per core and number of core after adding
+   * buffer
+   *
+   * @param maxPeakJVMUsedMemoryPerCore
+   * @param core
+   * @return suggested executor memory
+   */
+  private long suggestExecutorMemory(long maxPeakJVMUsedMemoryPerCore, int core) {
+    return (maxPeakJVMUsedMemoryPerCore * (100 + EXECUTOR_MEMORY_BUFFER_PER_CORE) / 100) * core
+        * (100 + EXECUTOR_MEMORY_BUFFER_OVERALL) / 100;
+  }
+
+  /**
+   * Suggest memory factor using peak unified memory per core and number of core after adding buffer
+   */
+  private void suggestMemoryFactor() {
+    long unifiedMemoryRequirement =
+        (getMaxPeakUnifiedMemoryPerCore() * (100 + EXECUTOR_MEMORY_BUFFER_PER_CORE) / 100) * suggestedCore
+            * (100 + EXECUTOR_MEMORY_BUFFER_OVERALL) / 100;
+    suggestedMemoryFactor = unifiedMemoryRequirement * 1.0 / suggestedExecutorMemory;
+  }
+
+  /**
+   * Suggest driver memory based on driver peak JVM used memory
+   */
+  private void suggestDriverMemory() {
+    suggestedDriverMemory = driverMaxPeakJVMUsedMemory * (100 + DRIVER_MEMORY_BUFFER) / 100;
+    if (suggestedDriverMemory < MIN_DRIVER_MEMORY) {
+      suggestedDriverMemory = MIN_DRIVER_MEMORY;
+    }
+    suggestedDriverMemory = getRoundedDriverMemory();
+  }
+
+  /**
+   * Rounded container size in multiple of 1GB
+   *
+   * @param memory
+   * @return rounded container size
+   */
+  private long getRoundedContainerSize(long memory) {
+    return ((long) Math.ceil(memory * 1.0 / FileUtils.ONE_GB)) * FileUtils.ONE_GB;
+  }
+
+  private int getMemoryIncreaseForGC() {
```

Comment:
There's a lot of comments about GC getting flagged. I'm inclined to be cautious, and give a good range for acceptable GC, and not decrease executor memory if GC time is low (decreasing executor memory would just be based on max JVM used memory).  5% increase in memory is pretty modest -- it may not be enough to resolve memory issues.

Reply:
I have removed the decrease executor memory part. We will keep it 5% for other case and see how it behaves with other jobs. We couldn't find jobs which have GC time more than threshold. There are very few production jobs which have GC time more than threshold.